### PR TITLE
Further updates to Java plugin process

### PR DESCRIPTION
**Pull request by kodebach**
Continuation of #4239 

## Basics

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [x] The PR is rebased with current master.

## Checklist

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

- [ ] Add the "work in progress" label if you do not want the PR to be reviewed yet.
- [ ] Add the "ready to merge" label **if the basics are fulfilled** and no further pushes are planned by you.

### CODE REVIEW

File: doc/help/kdb-mount-java.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+```
+
+This internally expands to (something like) this complicated snippet:
+
+```sh
+KDB_VERSION="$(kdb --version | sed -nE 's/KDB_VERSION: (.+)/\1/gp')"
+sudo kdb mount config.file user:/tests/process dump process "executable=$(command -v java)" 'args=#3' 'args/#0=-cp' "args/#1=/usr/share/java/libelektra-$KDB_VERSION-all.jar:/usr/share/java/process-$KDB_VERSION.jar:/usr/share/java/whitelist-$KDB_VERSION.jar" 'args/#2=org.libelektra.process.ProcessApp' 'args/#3=org.libelektra.plugin.WhitelistPlugin' type
+```
+
+As you can see, the every argument for a Java plugin is replaced by 7 (quite long) arguments.
```
Comment (markus2330) [inline doc/help/kdb-mount-java.md:35]:
```suggestion
As you can see, every argument for a Java plugin is replaced by several arguments.
```

File: doc/help/kdb-mount-java.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+Even worse, the arguments depend on how Elektra is installed on your system.
+The `mount-java` knows where Elektra's JARs are installed (`/usr/share/java` above) and constructs a classpath argument for the JVM.
+It also chooses the `ProcessApp` class as the main class and finally instructs this class to load the `WhitelistPlugin`.
```
Comment (markus2330) [inline doc/help/kdb-mount-java.md:38]: Other name for ProcessApp?

File: doc/help/kdb-mount-java.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+However, `mount-java` isn't magic and it only works this smoothly for plugins that come bundled with Elektra.
+If you have external plugins, you must also specify the additional classpath for these plugins like this:
+
+```sh
+CLASSPATH=/path/to/foo.jar:/path/to/bar.jar sudo kdb mount-java config.file user:/tests/foobar java:org.example.Foo java:org.example.Bar java:org.libelektra.plugin.WhitelistPlugin
```
Comment (markus2330) [inline doc/help/kdb-mount-java.md:44]:
```suggestion
export CLASSPATH=/path/to/foo.jar:/path/to/bar.jar
sudo --preserve-env=CLASSPATH kdb mount-java config.file user:/tests/foobar java:org.example.Foo java:org.example.Bar java:org.libelektra.plugin.WhitelistPlugin
```

File: src/plugins/dump/dump.cpp [hunk 4]
```diff
@@ -524,18 +524,18 @@ class FileStreamBuf : public std::streambuf
...
+	return unserialize (is, errorKey, ks, true);
 }
 
-int fwriteks (KeySet * ks, FILE * file, Key * errorKey)
+int fserialize (KeySet * ks, FILE * file, Key * errorKey)
 {
 	FileStreamBuf buf (file);
 	std::ostream os (&buf);
-	return serialise (os, errorKey, ks, true);
+	return serialize (os, errorKey, ks, true);
```
Comment (markus2330) [inline src/plugins/dump/dump.cpp:538]: Please add this change also in release notes. Also write a bit more about how to use Java plugins there. Can you also add

**Review by markus2330 — APPROVED** [review]
Thank you for the quick update!

File: src/plugins/dump/dump.cpp (hunk 4 above)
Comment (kodebach) [reply src/plugins/dump/dump.cpp:538]: I think part of the last sentence is missing...

Comment (markus2330) [reply src/plugins/dump/dump.cpp:538]: Yes, it was also about something in the release notes.

Comment (markus2330) [reply src/plugins/dump/dump.cpp:538]: As this is hopefully the last PR before the release, please see what you can do to improve the release notes.

Comment (kodebach) [reply src/plugins/dump/dump.cpp:538]: I have already added some stuff (Highlights, dump)

**Comment by markus2330** [comment]
Can you also pump the version in debian/changelog (see #4255)?

File: doc/news/_preparation_next_release.md [hunk 5]
```diff
@@ -81,6 +90,14 @@ The following section lists news about the [plugins](https://www.libelektra.org/
 
 - Fixed an issue with type-limits on ARM32 (see issue #4217). _(Klemens Böswirth @kodebach)_
 
+### dump
+
+- The exported functions `serialise` and `unserialise` have been renamed to `serialize` and `unserialize`.
+  For now the old version are still exported as an alias.
```
Comment (markus2330) [inline doc/news/_preparation_next_release.md:96]: I don't think we should do this as this is already preparation for 1.0.

File: doc/news/_preparation_next_release.md [hunk 6]
```diff
@@ -81,6 +90,14 @@ The following section lists news about the [plugins](https://www.libelektra.org/
...
+  This may be removed in a future version. _(Klemens Böswirth @kodebach)_
+- New exported functions `int fserialise(KeySet * ks, FILE * file, Key * errorKey)` and `int funserialise(KeySet * ks, FILE * file, Key * errorKey)` have been added.
```
Comment (markus2330) [inline doc/news/_preparation_next_release.md:98]:
```suggestion
- New exported functions `int fserialize(KeySet * ks, FILE * file, Key * errorKey)` and `int funserialize(KeySet * ks, FILE * file, Key * errorKey)` have been added.
```

File: doc/news/_preparation_next_release.md [hunk 7]
```diff
@@ -81,6 +90,14 @@ The following section lists news about the [plugins](https://www.libelektra.org/
...
+  These are wrappers around `serialise` and `unserialise` that allow calling from C with a standard `FILE *`. _(Klemens Böswirth @kodebach)_
```
Comment (markus2330) [inline doc/news/_preparation_next_release.md:99]:
```suggestion
  These are wrappers around `serialize` and `unserialize` that allow calling from C with a standard `FILE *`. _(Klemens Böswirth @kodebach)_
```

**Comment by kodebach** [comment]
> Can you also pump the version in debian/changelog (see #4255)?

done

**Review by markus2330 — APPROVED** [review]
Release notes already look much better!

File: scripts/packaging/debian/changelog [hunk 8]
```diff
@@ -1,3 +1,9 @@
+elektra (0.9.8-5) unstable; urgency=medium
+
+  * New upstream version.
```
Comment (markus2330) [inline scripts/packaging/debian/changelog:3]:
```suggestion
  * Pump version to make new packages visible #4255.
```

**Comment by kodebach** [comment]
I think this is now finally done...

**Pull request merged**